File: Cake.cpp
//
// Created by NandM2 on 12/10/2021.
//

#include "Cake.h"

#include <utility>

Cake::Cake(double basePrice, std::string flavor, std::string frosting) : BakedGood(basePrice), flavor(std::move(flavor)), frosting(std::move(frosting)) {}

const std::string &Cake::getFlavor() const {
    return flavor;
}

const std::string &Cake::getFrosting() const {
    return frosting;
}

File: Dart/bakery_checkout/bin/cake.dart
import 'baked_good.dart';

abstract class Cake extends BakedGood {
  final String _flavor, _frosting;

  Cake(double basePrice, this._flavor, this._frosting) : super(basePrice);

  String get flavor => _flavor;

  String get frosting => _frosting;
}
